micro_sam/evaluation/inference.py

```diff
@@ -395,44 +394,47 @@ def _save_segmentation(masks, prediction_path):
     imageio.imwrite(prediction_path, segmentation)
 
 
+def extract_instances_from_batched_outputs(batched_outputs):
+    masks = [
+        torch.stack(
+            [torch.sigmoid(_m[torch.argmax(_iou)][None]) for _m, _iou in zip(m["masks"], m["iou_predictions"])]
+        ) for m in batched_outputs
+    ]
+    masks = torch.stack(masks)
+    masks = (masks > 0.5).to(torch.float32)
+    return masks
+
+
+@torch.no_grad()
 def _run_inference_with_iterative_prompting_for_image(
-    model,
-    image,
-    gt,
-    n_iterations,
-    device,
-    use_boxes,
-    prediction_paths,
-    batch_size,
-):
+    model, image, gt, n_iterations, device, use_boxes, prediction_paths, batch_size
+) -> None:
     assert len(prediction_paths) == n_iterations, f"{len(prediction_paths)}, {n_iterations}"
-    to_sam_inputs = ConvertToSamInputs()
+    convert_to_sam_inputs = ConvertToSamInputs()
 
-    image = torch.from_numpy(
-        image[None, None] if image.ndim == 2 else image[None]
-    )
+    image = torch.from_numpy(image[None, None] if image.ndim == 2 else image[None])
     gt = torch.from_numpy(gt[None].astype("int32"))
 
     n_pos = 0 if use_boxes else 1
-    batched_inputs, sampled_ids = to_sam_inputs(image, gt, n_pos=n_pos, n_neg=0, get_boxes=use_boxes)
+    batched_inputs, sampled_ids = convert_to_sam_inputs(image, gt, n_pos=n_pos, n_neg=0, get_boxes=use_boxes)
 
     input_images = torch.stack([model.preprocess(x=x["image"].to(device)) for x in batched_inputs], dim=0)
     image_embeddings = model.image_embeddings_oft(input_images)
 
-    multimasking = n_pos == 1
+    multimasking = (n_pos == 1)
     prompt_generator = IterativePromptGenerator()
 
     n_samples = len(sampled_ids[0])
     n_batches = int(np.ceil(float(n_samples) / batch_size))
 
     for iteration in range(n_iterations):
-        final_masks = []
+        final_masks, all_sampled_binary_y = [], []
         for batch_idx in range(n_batches):
```

Comment:
You should use https://github.com/computational-cell-analytics/micro-sam/blob/dev/micro_sam/inference.py#L17 here.